### DISCUSSION TITLE
Setting up a carousel based on an array

### DISCUSSION THREAD

**Discussion by paulmiddelkoopux** [quasarframework/quasar#10936] [General - Components / Directives / etc] [answered]
Hi all, thank you for taking the time to look at my question. Please let me know if this is not the right place for this kind of question.

I am trying to show a carousel of items (in this case compliments, which have 'content' and a 'date'). For each item I want it to load in the carousel.  I guess I am not understanding how it exactly works. I tried to copy code from [this example](https://forum.quasar-framework.org/topic/7765/q-carousel-v-model-in-nested-v-for/5). Can anybody point out what I am getting wrong?

Edit: I made a [codepen](https://codepen.io/paulmiddelkoopux/pen/gOxOLGY)

```
<template>
  <q-page class="column bg-grey-3">
    <q-carousel 
      v-model="item.model"
      transition-prev="slide-right"
      transition-next="slide-left"
      swipeable
      animated
      :arrows="arrows"
      :padding="padding"
      control-color="dark"
      class="bg-white"
      >
          <q-carousel-slide 
            v-for="(content, i) in item.content"
            :key="i"
            :name='i'
            v-ripple
            class="row content-center">
            <div>
              {{ compliment.content }}
            </div>
            <div>
            {{ compliment.date }}
            </div>
          </q-carousel-slide>
    </q-carousel>
</template>


<script>
import { ref } from 'vue'

export default {
  setup () {
    return {
      arrows: ref(true),
      padding: ref(true),
      compliments: [
        {
          content:'Hey good lookin',
          date:'4 november 2012'
        },
        {
          content:'Arent you something',
          date:'12 mei 2011'
        },
        {
          content:'You are the best coder that is reading this right now',
          date:'5 oktober 1989'
        },
      ]
    }
  },
    created() {
    this.compliments = this.compliments.map(v => ({...v, model: 0}))
  }
}
</script>
```

**Comment by hawkeye64** [comment]
@paulmiddelkoopux I see a few issues, but it'd really help if you just put this into a CodePen and then when it's fixed, you will see the answer better than posting code here.

**Reply by paulmiddelkoopux** [reply]
Thanks for your reply, here is the codepen: https://codepen.io/paulmiddelkoopux/pen/gOxOLGY

**Comment by hawkeye64** [comment, marked as answer]
There was a lot of "extra" stuff in there that I removed, plus some things were wrong, so I inferred them. I am hoping this is what you wanted: https://codepen.io/Hawkeye64/pen/OJjJWMb?editors=1010

**Reply by paulmiddelkoopux** [reply]
Hi Jeff, thanks a ton! This is exactly the desired outcome. Cheers! And great that I can study what you did and learn from it 👍
